### ISSUE TITLE
Compatibility checks should be skipped during hot reload

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#727]
While they don't take much time, compatibility checks should still be completely skipped during hot reload. This can likely be achieved through active scene checking.

Ideally compatibilty classes should be changed from static to instance-based (see also: #211), and completely bypass firing up the instances.

@kianzarrin Any comments on detecting in-game hotload?

**Comment by kianzarrin** [comment]
> Any comments on detecting in-game hotload?

There are several ways of checking whether we are in game or not each of them with minor differences. Those minor differences don't matter as far as checking for incompatible mods are concerned. So both of these methods work:

1. One way as you said is to check the scene.
2. Another way is to check LoadingExtension.InGame (which checks app mode)

I suspect 99% of the hot-reload time is spent on Harmony.

**Comment by aubergine10** [comment]
Yup. I've decided to keep static methods and stuff for now as it's just so much simpler than managing instance lifecycle. But the compatibility checks will now be skipped if a hot-reload occurs in-game.

**Comment by aubergine10** [comment]
```
Debug 827.3824706: CompatibilityManager.Activate() Scene = Game
Debug 827.3830747: - Skipping due to in-game hot reload
```
